### PR TITLE
Small wrapper with the ability to add custom info in any position

### PR DESCRIPTION
**Pull request by ilya-bystrov**
The idea of this wrapper is to add custom info in any position.
I didn't find anything similar in `contrib` directory.

### CODE REVIEW

File: contrib/wrapper.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+# This i3status wrapper allows to add custom information in any position of the bar
+# It was developed for i3bar (JSON format)
+
+# The idea is to define "holder" modules in i3status config and than replace them
```
Comment (Airblader) [inline contrib/wrapper.sh]: s/than/then/

File: contrib/wrapper.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+# In order to make this example works you need to add
```
Comment (Airblader) [inline contrib/wrapper.sh]: s/works/work/

File: contrib/wrapper.sh [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# order += "tztime holder__hey_man"
+# and 
+# tztime holder__hey_man {
+#        format = "holder__hey_man"
+# }
+# in i3staus config 
+
+# Don't forget that
```
Comment (Airblader) [inline contrib/wrapper.sh]: Why is there a line break after just three words?

File: contrib/wrapper.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```
Comment (Airblader) [inline contrib/wrapper.sh]: Can we give this a more specific name than `wrapper.sh`? Something that conveys what it does would be good.

I know two other scripts in here are already named wrapper, but they're seven years old and if anything it proves my point... :-)

File: contrib/wrapper.sh [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+# This i3status wrapper allows to add custom information in any position of the bar
```
Comment (Airblader) [inline contrib/wrapper.sh]: In any position of the *statusline*, not the bar (you cannot place something left of the workspaces with this).

File: contrib/wrapper.sh [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+# line
+
+function update_holder {
+
+  local instance="$1"
+  local replacement="{ \"full_text\": \"$2\" }"
+  echo "$json_array" | jq --argjson arg_j "$replacement" "(.[] | (select(.instance==\"$instance\"))) |= \$arg_j" 
+}
+
+i3status | (read line; echo "$line"; read line ; echo "$line" ; read line ; echo "$line" ; while true
```
Comment (Airblader) [inline contrib/wrapper.sh]: Why are we read-echoing three times here? Shouldn't it just be twice (version + start of array)?

Comment (ilya-bystrov) [reply contrib/wrapper.sh]: Here it is. Have to wait for comma prefixed JSON-array.

```
$ i3status | while true; do read line ; echo $line ; echo '==='; done
{"version":1}
===
[
===
[{"name":"disk_info","instance":"/","markup":"none","full_text":"45.2 GiB"}]
===
,[{"name":"disk_info","instance":"/","markup":"none","full_text":"45.2 GiB"}]
===
,[{"name":"disk_info","instance":"/","markup":"none","full_text":"45.2 GiB"}]
===
```

File: contrib/wrapper.sh (hunk 4 above)
Comment (ilya-bystrov) [reply contrib/wrapper.sh]: My intention was to be able to place custom module in any position.
So I suggest `any_position_wrapper.sh`

**Comment by ilya-bystrov** [comment]
> Thanks! As a final step, could you please squash the commits into a single one?
Sure, I've done it

**Comment by ilya-bystrov** [comment]
@Airblader up

**Comment by Airblader** [comment]
Thank you! I apologize for the delay :-)

**Pull request merged**